ddtrace/contrib/pytest/_efd_utils.py

```diff
@@ -164,48 +158,113 @@ def _efd_get_outcome_from_item(
     return _TestOutcome(status=_outcome_status, skip_reason=_outcome_skip_reason, exc_info=_outcome_exc_info)
 
 
-def efd_pytest_terminal_summary(terminalreporter):
+def efd_pytest_terminal_summary_post_yield(terminalreporter: _pytest.terminal.TerminalReporter):
     terminalreporter.write_sep("=", "Datadog Early Flake Detection", purple=True, bold=True)
     # Print summary info
-    passed_reports = terminalreporter.getreports(_EFD_RETRY_OUTCOMES.EFD_FINAL_PASSED)
-    if passed_reports:
-        terminalreporter.write_sep("_", "PASSED", green=True, bold=True)
-        for passed_report in passed_reports:
-            line = f"{terminalreporter._tw.markup('PASSED', green=True)} {passed_report.nodeid}"
-            terminalreporter.write_line(line)
-        del terminalreporter.stats[_EFD_RETRY_OUTCOMES.EFD_FINAL_PASSED]
+    raw_summary_strings = []
+    markedup_summary_strings = []
+
     failed_reports = terminalreporter.getreports(_EFD_RETRY_OUTCOMES.EFD_FINAL_FAILED)
     if failed_reports:
+        failed_text = f"{len(failed_reports)} failed"
+        raw_summary_strings.append(failed_text)
+        markedup_summary_strings.append(terminalreporter._tw.markup(failed_text, red=True, bold=True))
         terminalreporter.write_sep("_", "FAILED", red=True, bold=True)
         for failed_report in failed_reports:
             line = f"{terminalreporter._tw.markup('FAILED', red=True)} {failed_report.nodeid}"
             terminalreporter.write_line(line)
+            failed_report.outcome = PYTEST_STATUS.FAILED
+            terminalreporter.stats.setdefault(PYTEST_STATUS.FAILED, []).append(failed_report)
+
         del terminalreporter.stats[_EFD_RETRY_OUTCOMES.EFD_FINAL_FAILED]
+
+    passed_reports = terminalreporter.getreports(_EFD_RETRY_OUTCOMES.EFD_FINAL_PASSED)
+    if passed_reports:
+        passed_text = f"{len(passed_reports)} passed"
+        raw_summary_strings.append(passed_text)
+        markedup_summary_strings.append(terminalreporter._tw.markup(passed_text, green=True, bold=False))
+        terminalreporter.write_sep("_", "PASSED", green=True, bold=True)
+        for passed_report in passed_reports:
+            line = f"{terminalreporter._tw.markup('PASSED', green=True)} {passed_report.nodeid}"
+            terminalreporter.write_line(line)
+            passed_report.outcome = PYTEST_STATUS.PASSED
+            terminalreporter.stats.setdefault(PYTEST_STATUS.PASSED, []).append(passed_report)
+        del terminalreporter.stats[_EFD_RETRY_OUTCOMES.EFD_FINAL_PASSED]
```

Comment:
You mentioned it's hard to turn this into a function because then you have to pass a lot of arguments, but maybe we could write an internal function instead?